mypy/messages.py

```diff
@@ -570,9 +570,23 @@ def too_many_positional_arguments(self, callee: CallableType,
         msg = 'Too many positional arguments' + for_function(callee)
         self.fail(msg, context)
 
-    def unexpected_keyword_argument(self, callee: CallableType, name: str,
+    def unexpected_keyword_argument(self, callee: CallableType, name: str, arg_type: Type,
                                     context: Context) -> None:
         msg = 'Unexpected keyword argument "{}"'.format(name) + for_function(callee)
+        # Suggest intended keyword, look for same type match else check for any match.
+        matches = []
+        matching_type_args = []
+        for i, kwarg_type in enumerate(callee.arg_types):
+            callee_arg_name = callee.arg_names[i]
+            if (callee_arg_name is not None and
+                    (is_same_type(arg_type, kwarg_type) or is_subtype(arg_type, kwarg_type))):
```

Comment:
You don't need the `is_same_type()` here,`is_subtype(X, X)` returns `True`.

Reply:
Done